pyengine/pathfinder.py

```diff
@@ -21,4 +23,32 @@ def pathfinder(routes):
             return singledest['destination']
         else:
             print('pathfinder: no destinations available')
-            return False
\ No newline at end of file
+            return False
+            
+    elif routesparse['type'] == 'priority':
+        print('pathfinder: priority destination routing')
+        priorityroutes = []
+        for destination in routesparse['destinations']:
+            
+            # do not use routes which are not listed as available
+            if destination['available'] == 'yes':
+                priorityroutes.append(destination)
+            else:
+                print('pathfinder: {} is not available'.format(destination['destination']))
+            
+        # at this point, if there are no destinations available, we cannot use anything
+        if len(priorityroutes) == 0:
+            print('pathfinder: no destinations available!')
+            return False
+
+        bestpriority = 9999
```

Comment:
would not work if there are more than 9999 destinations, which i suppose there would not be